openmetadata-service/src/main/java/org/openmetadata/service/search/elasticsearch/ElasticSearchClientImpl.java

```diff
@@ -1011,27 +1018,17 @@ public void updateSearchEntityDeleted(EntityInterface entity, String scriptTxt,
   }
 
   @Override
-  public void deleteEntityAndRemoveRelationships(EntityInterface entity, String scriptTxt, String field) {
+  public void softDeleteOrRestoreEntityFromSearch(EntityInterface entity, boolean delete) {
     if (entity != null) {
       String entityType = entity.getEntityReference().getType();
       SearchIndexDefinition.ElasticSearchIndexType indexType = IndexUtil.getIndexMappingByEntityType(entityType);
-      DeleteRequest deleteRequest = new DeleteRequest(indexType.indexName, entity.getId().toString());
-      deleteRequest.setRefreshPolicy(WriteRequest.RefreshPolicy.IMMEDIATE);
+      UpdateRequest updateRequest = new UpdateRequest(indexType.indexName, entity.getId().toString());
+      String scriptTxt = "ctx._source.deleted=" + delete;
```

Comment:
we should declare all the scriptTexts in one place and call this softDeleteOrRestoreScriptTxt